driver/src/main/com/mongodb/reactivestreams/client/internal/GridFSDownloadPublisherImpl.java

```diff
@@ -127,6 +127,7 @@ public void onComplete() {
                     synchronized (GridFSDownloadSubscription.this) {
                         hasTerminated = currentAction == Action.TERMINATE || currentAction == Action.FINISHED;
                         if (!hasTerminated) {
+                            requested--;
```

Comment:
requested - flags the user demand. As this is a Single Result Publisher it could be a boolean (eg hasRequested data) but keeping the count is inline with other publishers.

The problem this fixes was that requested count was used as the request demand for the source publisher. That is fine if like in the tests we automatically request Integer.MAX_VALUE.  

However, this publisher should return a fileId if request is 1 (which it didn't because previously we decremented requested in the wrong place).